grpc-sys/src/lib.rs

```diff
@@ -97,33 +169,83 @@ impl From<i32> for GrpcStatusCode {
     }
 }
 
+/// Result of a gRPC call.
+///
+/// If the caller satisfies the prerequisites of a
+/// particular operation, the `GrpcCallStatus` returned will be `Ok`.
+/// Receiving any other value listed here is an indication of a bug in the caller.
+///
+/// Based on `grpc_call_error`.
 #[repr(C)]
 #[derive(Debug, PartialEq)]
 pub enum GrpcCallStatus {
+    /// Everything went ok.
     Ok = 0,
+
+    /// Something failed, we don't know what.
     Error,
+
+    /// This method is not available on the server.
     ErrorNotOnServer,
+
+    /// This method is not available on the client.
     ErrorNotOnClient,
+
+    /// This method must be called before server_accept.
     ErrorAlreadyAccepted,
+
+    /// This method must be called before invoke.
     ErrorAlreadyInvoked,
+
+    /// This method must be called after invoke.
     ErrorNotInvoked,
+
+    /// This call is already finished (writes_done or write_status has already been called).
     ErrorAlreadyFinished,
+
+    /// There is already an outstanding read/write operation on the call.
     ErrorTooManyOperations,
+
+    /// The flags value was illegal for this call.
     ErrorInvalidFlags,
+
+    /// Invalid metadata was passed to this call.
     ErrorInvalidMetadata,
+
+    /// Invalid message was passed to this call.
     ErrorInvalidMessage,
+
+    /// Completion queue for notification has not been registered with the server.
     ErrorNotServerCompletionQueue,
+
+    /// This batch of operations leads to more operations than allowed.
     ErrorBatchTooBig,
+
+    /// Payload type requested is not the type registered.
     ErrorPayloadTypeMismatch,
+
+    /// Completion queue has been shut down.
+    ErrorCompletionQueueShutdown,
 }
 
+/// The type of completion.
+///
+/// Based on `grpc_completion_type`.
 #[repr(C)]
 pub enum GrpcCompletionType {
+    /// Shutting down.
     QueueShutdown,
+
+    /// No event before timeout
```

Comment:
You do not have periods on the end of the comments here and below when you do for the others.

Reply:
nice catch!